clang/lib/CIR/CodeGen/CIRGenStmtOpenACC.cpp

```diff
@@ -109,14 +126,15 @@ class OpenACCClauseCIREmitter final
   }
 
   void VisitDeviceTypeClause(const OpenACCDeviceTypeClause &clause) {
+    lastDeviceTypeClause = &clause;
```

Comment:
What's the lifetime of this object? Who owns it?

Reply:
Its owned by the AST, so its lifetime lasts until the AST is destroyed, which is after frontend codegen.